modules/message-lane/src/lib.rs

```diff
@@ -0,0 +1,283 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Runtime module that allows sending and receiving messages using lane concept:
+//!
+//! 1) the message is sent using `send_message()` call;
+//! 2) every outbound message is assigned nonce;
+//! 3) the messages are stored in the storage;
+//! 4) external component (relay) delivers messages to bridged chain;
+//! 5) messages are processed in order (ordered by assigned nonce);
+//! 6) relay may send proof-of-receiving and proof-of-processing back to this chain.
+//!
+//! Once message is sent, its progress can be tracked by looking at module events.
+//! The assigned nonce is reported using `MessageAccepted` event. When message is
+//! accepted by the bridged chain, `MessagesDelivered` is fired. When message is
+//! processedby the bridged chain, `MessagesProcessed` by the bridged chain.
+
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
```suggestion
#![cfg_attr(not(feature = "std"), no_std)]
#![warn(missing_docs)]
```

Reply:
I've added this, but now it shows warning about undocumented struct. I believe it comes somewhere from decl_module, or decl_storage macros. So I've reverted this for now

Reply:
Have checked Substrate - we do not have this attr in pallets there either :)